### ISSUE TITLE
 Exception while invoking method 'createUser' TypeError: Cannot create property 'deleted'

### ISSUE THREAD

**Issue by guncebektas** [jamauro/soft-delete#1]
```
 Exception while invoking method 'createUser' TypeError: Cannot create property 'deleted' on string 'aZ23isxtE7L5QEoj8'
I20240912-10:49:24.411(3)?     at addDeleted (packages/jam:soft-delete/soft-delete.js:59:27)
I20240912-10:49:24.411(3)?     at Collection.Mongo.Collection.<computed> [as findOneAsync] (packages/jam:soft-delete/soft-delete.js:131:9)
I20240912-10:49:24.411(3)?     at AccountsServer._attemptLogin (packages/accounts-base/accounts_server.js:458:31)
I20240912-10:49:24.411(3)?     at AccountsServer._loginMethod (packages/accounts-base/accounts_server.js:509:23)
I20240912-10:49:24.411(3)?     at processTicksAndRejections (node:internal/process/task_queues:95:5)
I20240912-10:49:24.411(3)?     at MethodInvocation.createUser (packages/accounts-password/password_server.js:1004:14)
I20240912-10:49:24.411(3)?     at Session.<anonymous> (packages/montiapm:agent/lib/hijack/wrap_session.js:77:18)
```

Hi Jam, I got a warning while trying to log in or create a user. If I remove your package everything works fine, can you please check it. thanks.

**Comment by harry-73** [comment]
I have the same issue when I use `Meteor.loginWithPassword`. Without the package, all is fine. 

With the package, two different cases:

- without adding 'deleted' field to the users collection:

I got:
```
{
    "isClientSafe": true,
    "error": 403,
    "reason": "error.accounts.Login forbidden",
    "message": "error.accounts.Login forbidden [403]",
    "errorType": "Meteor.Error"
}
```

- with adding 'deleted' field to the users collection:

I have the same error on client side and have that one on server side:

```
I20240912-10:32:06.956(2)? Exception while invoking method 'login' TypeError: Cannot create property 'deleted' on string 'ack5L5ivM5ACBeNpK'
I20240912-10:32:06.957(2)?     at addDeleted (packages/jam:soft-delete1/soft-delete.js:59:27)
I20240912-10:32:06.957(2)?     at Collection.Mongo.Collection.<computed> [as findOneAsync] (packages/jam:soft-delete1/soft-delete.js:131:9)
I20240912-10:32:06.957(2)?     at AccountsServer._attemptLogin (packages/accounts-base/accounts_server.js:458:31)
I20240912-10:32:06.958(2)?     at MethodInvocation.methods.login (packages/accounts-base/accounts_server.js:657:29)
I20240912-10:32:06.958(2)?     at Session.<anonymous> (packages/montiapm:agent/lib/hijack/wrap_session.js:75:18)

```

**Comment by guncebektas** [comment]
I don't really like working with the Users collection. I think there should be an exception defined for this collection. In fact, we should be able to define exceptions for any collections we want. Scenarios like activating or deactivating users in the Users collection shouldn't be part of this package.

**Comment by jamauro** [comment]
Thanks for reporting. This issue has been fixed in `v0.1.1`

**Comment by guncebektas** [comment]
The user registration feature is working, but the login process returns a "User not found" error. You can find a reproduction of the issue here: https://github.com/guncebektas/saas-boilerplate. I'm currently using your excellent packages and am happy to have them as part of my project. The project hasn't been announced yet, but feel free to use it for reproduction.

**Comment by jamauro** [comment]
I took a quick look at your repo. I didn't see a `deleted: false` flag for any of your Collection docs. This package assumes that flag will be added by you, e.g. on an insert or as a default value as part of your schema. At the moment, it does not add it to the document automatically, but it will add that condition automatically on your queries, e.g. `.find`. I'm assuming this is why you're encountering an issue. 

Let me know if you have any thoughts on whether this package should add it to the document automatically. I'll have to think on that a bit.

**Comment by harry-73** [comment]
The issue I have is that `uId = await Accounts.createUserAsync` creates a user but without the field `delete: false`.

So When I tried to find the user with `Meteor.users.findOneAsync({ _id: uId })` , it failed.

**Comment by guncebektas** [comment]
thank you for your package but can we really call this zero-config? I don't understand what to do with the users table right now.

**Comment by jamauro** [comment]
Fair point, thanks for the feedback. I plan on addressing this soon.

**Comment by jamauro** [comment]
This has been fixed in `v0.2.0` 🎉 . It will automatically add the `deleted` flag on user creation and on insert for other collections. I'll close this issue but if you encounter any problems, feel free to reopen.

**Comment by guncebektas** [comment]
Hi, now I can't call onCreateUser in my code

`
 Error: Can only call onCreateUser once
I20240923-11:53:42.369(3)?     at AccountsServer.onCreateUser (packages/accounts-base/accounts_server.js:193:13)
I20240923-11:53:42.369(3)?     at packages/jam:soft-delete/soft-delete.js:126:14
I20240923-11:53:42.369(3)?     at Function.time (/Users/gunce/Dev/saas-boilerplate/.meteor/local/build/programs/server/tools/tool-env/profile.ts:310:30)
I20240923-11:53:42.369(3)?     at /tools/static-assets/server/boot.js:453:19
I20240923-11:53:42.369(3)?     at processTicksAndRejections (node:internal/process/task_queues:95:5)
I20240923-11:53:42.369(3)?     at /tools/static-assets/server/boot.js:504:7
I20240923-11:53:42.369(3)?     at startServerProcess (/tools/static-assets/server/boot.js:501:3)

`

**Comment by jamauro** [comment]
Thanks for letting me know. Apologies for the trouble. I've published `v0.2.1` to address this.
